Providers/PythonProvider.cpp

```diff
@@ -48,22 +49,51 @@ char const SCRIPT_PATH_EXTENSION[] = "/lib/Scripts/";
 char const DEFAULT_DSC_SCRIPT[] = "client";
 char const PY_EXTENSION[] = ".py";
 
+std::string determinePythonVersion(){
+    std::array<char, 128> buffer;
+    std::string result;
+    // Check for python2
+    FILE* pipe = popen("python2 --version 2>&1", "r");
```

Comment:
will checking for python2 work ?

Reply:
Yes, ideally both will result in the same outcome. but to have a better scenario coverage I will update this to python instead of python2 at all there files where similar code is implemented (.sh, .c, .cpp). Good catch!